src/components/Visualization.tsx

```diff
@@ -51,10 +53,55 @@ const Visualization = () => {
     shallow,
   );
 
-  const flows = useFlowsStore((state) => state.flows);
+  const { settings, setSettings } = useSettingsStore((state) => state);
+  const [ setSourceCode, setSyncedSourceCode ] = useIntegrationSourceStore(
+    (state) => [state.setSourceCode, state.setSyncedSourceCode], shallow);
+  const { flows, properties, metadata } = useFlowsStore(
+    ({ flows, properties, metadata }) => ({
+      flows,
+      properties,
+      metadata,
+    }), shallow
+  );
   const visualizationService = useMemo(() => new VisualizationService(), []);
   const stepsService = useMemo(() => new StepsService(), []);
 
+  const previousFlows = usePrevious(flows);
+
+  useEffect(() => {
+    if (previousFlows === flows || !Array.isArray(flows[0].steps)) return;
```

Comment:
This will cause an error whenever we delete all existing flows by trying to access `undefined`. [It happened to me in the past](https://github.com/KaotoIO/kaoto-ui/pull/1927/files#diff-adf7a8db55b4b1624ee2eaf3c331f474481bde7d8e79927dfec164216372bb99L45) :sweat_smile:.

```suggestion
    if (previousFlows === flows || !Array.isArray(flows[0]?.steps)) return;
```

Reply:
thanks for review - updated